src/components/InvertedFlatList/BaseInvertedFlatList.js

```diff
@@ -96,22 +113,36 @@ class BaseInvertedFlatList extends Component {
     measureItemLayout(nativeEvent, index) {
         const computedHeight = nativeEvent.layout.height;
 
-        // We've already measured this item so we don't need to
-        // measure it again.
-        if (this.sizeMap[index]) {
+        // Unclear why but some items will read 0 for height when onLayout runs
+        if (computedHeight === 0) {
             return;
         }
 
-        const previousItem = this.sizeMap[index - 1] || {};
+        // We've already measured this item so we don't need to measure it again.
+        if (this.sizeMap[index]) {
+            return;
+        }
 
-        // If there is no previousItem this can mean we haven't yet measured
-        // the previous item or that we are at index 0 and there is no previousItem
-        const previousLength = previousItem.length || 0;
-        const previousOffset = previousItem.offset || 0;
         this.sizeMap[index] = {
             length: computedHeight,
-            offset: previousLength + previousOffset,
         };
+
+        if (_.size(this.sizeMap) === this.props.data.length) {
```

Comment:
From my testing of clicking 5 different chats in web, desktop and ios safari it hits this block. The largest `this.props.data.length` that i have seen passed is 172 and it worked fine,
![image](https://user-images.githubusercontent.com/14356145/169410957-b08e2879-6b58-4f10-b614-62865ec2d4b9.png)
(My log line was 
```
console.log(`over here sizeMap: ${_.size(this.sizeMap)} : ${this.props.data.length}`);
``` 
just before this line and as you can see the sizeMap count did reach props.data.length)

Reply:
So from this [comment](https://github.com/Expensify/App/pull/8638#discussion_r882907781) i only allow scroll back of 150 comments (it sometimes goes up to 200 depending on initial render of items). So i don't think we need to test against 1000 comments. I've also now added this to the "known issues/limitation" section in the PR description.

So far with scroll at position between 100-150 seems to been working fine. The scroll highlight sometimes may not be perfectly centered but i feel like that could be a tackled as a future improvement if needed. Let me know if you guys disagree.

Reply:
Hmm unless onyx has somehow locally stored 1000 comments and when we switch to the report this.props.data.length would be 1000 so maybe sizeMap won't render them all. Hmm okay i see what you guys mean. I'll test it out.

Reply:
Right, I guess in a world where we have bidirectional scrolling this would be less of a concern because we can just always make sure that we load up `n` comments and would have no problem rendering something like 50-100 comments. But like you said, it gets tricky when you have a lot of cached data. 

I am not sure what our current process for getting rid of old report data is (if we have one besides the cache eviction stuff - which should hardly happen on web anymore because we're using IndexedDB).

Reply:
So i just tested it with cached data on web and desktop. I tested it like so,
1. Wrote a query to inject 1100 comments to a specific report. This report now has 1200 comments in total.
2. I navigated to the report and scrolled to the top to load more comments. Kept scrolling to the top until every comment was seen and hence loaded onto onyx. I also had a console log to confirm all were loaded.
3. I clicked another report and then clicked back to this first report. Had a console log running to check how many sizeMaps were calculated and its length,
```
console.log(`over here sizeMap: ${_.size(this.sizeMap)} : ${this.props.data.length}`);
```
4. the data length was 1200 and the sizeMap was calculated for every entry that was cached without me scrolling.
![image](https://user-images.githubusercontent.com/14356145/171479457-d3ee2571-dcd6-4a7e-9494-9578cdb8219f.png)

It is terribly inefficient though since if i want to comment highlight something thats the most recent 50 i've to wait like 2 minutes for the sizeMap to completely load. 

But i think for this phase we don't have to worry about this because with bidirectional scrolling we dont have to worry about performance improvements while report switching since we will open the selected reportActionID directly. Let me know if thats okay with you guys.

Reply:
Yup, as of now i've mentioned it as part of "Known problems/limitations" in the PR description,
> Clicking on a comment link on another report takes a while for the animation to begin. Especially on web/desktop since it takes a while before we render and calculate all the heights.

I'll update it to mention it should be a non-issue once bidirectional scrolling is implement because clicking a comment link on another report will directly load the desired reportActionID first.